test/Microsoft.AspNetCore.Cors.Test/CorsPolicyExtensionsTests.cs

```diff
@@ -23,6 +23,25 @@ public void IsOriginAnAllowedSubdomain_ReturnsTrueIfPolicyContainsOrigin()
             Assert.True(actual);
         }
 
+        [Theory]
+        [InlineData(null)]
+        [InlineData("null")]
+        [InlineData("http://")]
+        [InlineData("http://*")]
+        [InlineData("http://.domain")]
+        public void IsOriginAnAllowedSubdomain_ReturnsFalseIfOriginIsMalformedUri(string malformedOrigin)
```

Comment:
Should we also test some URLs that have something after the authority portion? like: `http://foo.bar/kdkfdkfdkf`

Reply:
This test is specifically for malformed origins. Other tests cover well-formed Uris. I have added a test for the following: `"http://.domain/hello")]`